### PR TITLE
test:schedule_api: add ram filter

### PR DESCRIPTION
**Pull request by hakehuang**
the min_ram is about: 33KBytes and set to 36KBytes

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

File: tests/kernel/sched/schedule_api/testcase.yaml [hunk 1]
```diff
@@ -4,18 +4,22 @@ tests:
     extra_configs:
       - CONFIG_TIMESLICING=y
     tags: kernel threads sched userspace
+    min_ram: 36
   kernel.scheduler.no_timeslicing:
     filter: not CONFIG_SCHED_MULTIQ
     extra_configs:
       - CONFIG_TIMESLICING=n
     tags: kernel threads sched userspace
+    min_ram: 36
```
Comment (nashif) [inline tests/kernel/sched/schedule_api/testcase.yaml:13]: how did you come up with this number of 36?

Comment (hakehuang) [reply tests/kernel/sched/schedule_api/testcase.yaml:13]: the failure board has RAM=32, so I just enlage it to 36 to skip those failure boards.

Comment (nashif) [reply tests/kernel/sched/schedule_api/testcase.yaml:13]: well, we should not exclude tests from majority of boards based on 1 single board constraints. Overflows right now are being skipped in sanitycheck and should not be considered as failures, if a test does not fit in a board, it is not a bug.
Where are you getting those failures?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
NACK

This PR just has one commit, but it is targeting many different tests!!
Also, for example the mem_protect case I just built it and it is only using 22K for `mps2_an385`? 

```
[147/158] Linking C executable zephyr/zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       71376 B         4 MB      1.70%
            SRAM:       22036 B         4 MB      0.53%
        IDT_LIST:         200 B         2 KB      9.77%
[158/158] Linking C executable zephyr/zephyr.elf
```

File: tests/kernel/sched/schedule_api/testcase.yaml (hunk 1 above)
Comment (hakehuang) [reply tests/kernel/sched/schedule_api/testcase.yaml:13]: well, when I manuall check the kernel cases, I guess there shall be some clear explaination on why this case is skipped.

**Comment by hakehuang** [comment]
> NACK
> 
> This PR just has one commit, but it is targeting many different tests!!
> Also, for example the mem_protect case I just built it and it is only using 22K for `mps2_an385`?
> 
> ```
> [147/158] Linking C executable zephyr/zephyr_prebuilt.elf
> Memory region         Used Size  Region Size  %age Used
>            FLASH:       71376 B         4 MB      1.70%
>             SRAM:       22036 B         4 MB      0.53%
>         IDT_LIST:         200 B         2 KB      9.77%
> [158/158] Linking C executable zephyr/zephyr.elf
> ```

the board I test is mimxrt1015_evk.

**Pull request closed without merging**